x-pack/plugins/session_view/public/components/ProcessTree/index.tsx

```diff
@@ -0,0 +1,184 @@
+  const styles = useStyles();
+
+  const { sessionLeader, processMap, searchResults } = useProcessTree({
+    sessionEntityId,
+    data,
+    searchQuery,
+  });
+
+  const scrollerRef = useRef<HTMLDivElement>(null);
+  const selectionAreaRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (setSearchResults) {
+      setSearchResults(searchResults);
+    }
+  }, [searchResults, setSearchResults]);
+
+  useScroll({
+    div: scrollerRef.current,
+    handler: (pos: number, endReached: boolean) => {
+      if (!isFetching && endReached) {
+        fetchNextPage();
+      }
+    },
+  });
+
+  /**
+   * highlights a process in the tree
+   * we do it this way to avoid state changes on potentially thousands of <Process> components
+   */
+  const selectProcess = useCallback((process: Process) => {
+    if (!selectionAreaRef || !scrollerRef) {
+      return;
+    }
+
+    if (!selectionAreaRef.current || !scrollerRef.current) {
+      return;
+    }
+
+    const selectionAreaEl = selectionAreaRef.current;
+    selectionAreaEl.style.display = 'block';
+
+    // TODO: concept of alert level unknown wrt to elastic security
+    const alertLevel = process.getMaxAlertLevel();
+
+    if (alertLevel && alertLevel >= 0) {
+      selectionAreaEl.style.backgroundColor =
+        alertLevel > 0 ? 'rgba(229, 115, 115, 0.24)' : '#F2C94C4A';
+    } else {
+      selectionAreaEl.style.backgroundColor = '';
+    }
+
+    // find the DOM element for the command which is selected by id
+    const processEl = scrollerRef.current.querySelector<HTMLElement>(`[data-id="${process.id}"]`);
+
+    if (processEl) {
+      processEl.prepend(selectionAreaEl);
+
+      const cTop = scrollerRef.current.scrollTop;
+      const cBottom = cTop + scrollerRef.current.clientHeight;
+
+      const eTop = processEl.offsetTop;
+      const eBottom = eTop + processEl.clientHeight;
+      const isVisible = eTop >= cTop && eBottom <= cBottom;
+
+      if (!isVisible) {
+        processEl.scrollIntoView({ block: 'center' });
+      }
+    }
+  }, []);
+
+  useLayoutEffect(() => {
+    if (selectedProcess) {
+      selectProcess(selectedProcess);
+    }
+  }, [selectedProcess, selectProcess]);
+
+  useEffect(() => {
+    // after 2 pages are loaded (due to bi-directional jump to), auto select the process
+    // for the jumpToEvent
+    if (jumpToEvent && data.length === 2) {
+      const process = processMap[jumpToEvent.process.entity_id];
+
+      if (process) {
+        onProcessSelected(process);
+      }
+    }
+  }, [jumpToEvent, processMap, onProcessSelected, data]);
+
+  // auto selects the session leader process if no selection is made yet
+  useEffect(() => {
+    if (!selectedProcess) {
+      onProcessSelected(sessionLeader);
+    }
+  }, [sessionLeader, onProcessSelected, selectedProcess]);
+
+  function renderLoadMoreButton(text: JSX.Element, func: FetchFunction) {
+    return (
+      <EuiButton fullWidth onClick={() => func()} isLoading={isFetching}>
```

Comment:
Nit: Not sure how much of a performance benefit it will be, but this function is going to be recreated on every render. You could do an `const onClick = useCallback(() => func(), [func]);`, but I think it'll still recreate on every render because `renderLoadMoreButton` will be called on every render? It might be worth to just use the actual component in the body?